src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -399,15 +399,15 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                     return CheckPropertyValueKind(node, expr, valueKind, checkingReceiver, diagnostics);
 
                 case BoundKind.IndexOrRangePatternIndexerAccess:
-                    var patternIndexer = ((BoundIndexOrRangePatternIndexerAccess)expr);
-                    if (patternIndexer.PatternSymbol.Kind == SymbolKind.Property)
+                    var implicitIndexer = (BoundIndexOrRangePatternIndexerAccess)expr;
+                    if (implicitIndexer.IndexerAccess is BoundIndexerAccess implicitIndexerAccess)
                     {
                         // If this is an Index indexer, PatternSymbol should be a property, pointing to the
-                        // pattern indexer. If it's a Range access, it will be a method, pointing to a Slice method
+                        // implicit indexer. If it's a Range access, it will be a method, pointing to a Slice method
```

Comment:
>implicit [](http://example.com/codeflow?start=27&length=8)

I think this is the underlying indexer, not an implicit indexer, which doesn't exist as a real member.